packages/devtools/src/commands/elementClick.js

```diff
@@ -29,11 +30,43 @@ export default async function elementClick ({ elementId }) {
      * ensure to fulfill the click promise if the click has triggered an alert
      */
     return new Promise((resolve, reject) => {
+        let waitForPageLoadTimeout
+
+        /**
+         * check if page load has happened due to click
+         */
+        const frameNavigatedHandler = () => {
+            /**
+             * if so clear the `waitForPageLoadTimeout` to prevent command
+             * to finish and wait until the page has loaded
+             */
+            clearTimeout(waitForPageLoadTimeout)
+            page.once('load', () => resolve(null))
+        }
+        page.once('framenavigated', frameNavigatedHandler)
+
+        /**
+         * listen on possible modal dialogs that might pop up due to the
+         * click action, just continue in this case
+         */
         const dialogHandler = () => resolve()
         page.once('dialog', dialogHandler)
         return elementHandle.click().then(() => {
+            /**
+             * no modals popped up, so clean up the listener
+             */
             page.removeListener('dialog', dialogHandler)
-            resolve(null)
+
+            /**
+             * wait for at least 150ms to see if a page load was triggered,
```

Comment:
does it mean that we'll wait for 150ms after every click?

Reply:
Yes .. I know it is not ideal but the only way we can guarantee we have stable page transitions

Reply:
I don't know how to handle it better neither, we may improve it later on